gax-java/gax/src/test/java/com/google/api/gax/retrying/ScheduledRetryingExecutorTest.java

```diff
@@ -250,31 +251,47 @@ public void testCancelGetAttempt() throws Exception {
 
   @Test
   public void testCancelOuterFutureAfterStart() throws Exception {
+    ScheduledExecutorService localExecutor = Executors.newSingleThreadScheduledExecutor();
+    RetrySettings retrySettings =
+        FAST_RETRY_SETTINGS
+            .toBuilder()
+            // These params were selected to ensure that future tries to run and fail (at least
+            // once) but does not complete before it is cancelled. Assuming no computation time,
+            // it would take 25 + 50 + 100 + 100 = 275ms for the future to complete, which should
+            // be more than enough time to cancel the future.
+            .setInitialRetryDelay(Duration.ofMillis(25L))
+            .setMaxRetryDelay(Duration.ofMillis(100L))
+            .setRetryDelayMultiplier(2.0)
+            .setTotalTimeout(Duration.ofMillis(1000L))
+            // Set this test to not use jitter as the randomized retry delay (RRD) may introduce
+            // flaky results. For example, if every RRD value is calculated to be a small value
+            // (i.e. 2ms), four retries would result a "SUCCESS" result after 8ms, far below
+            // both the sleep value (50ms) and timeout (1000ms). This could potentially result
+            // in the future.cancel() returning false as you can't cancel a future that has
+            // already succeeded. The possibility of having each of the four retries produce a
+            // tiny RRD value is small, but not impossible.
+            .setJittered(false)
+            .build();
     for (int executionsCount = 0; executionsCount < EXECUTIONS_COUNT; executionsCount++) {
-      ScheduledExecutorService localExecutor = Executors.newSingleThreadScheduledExecutor();
-      FailingCallable callable = new FailingCallable(4, "requset", "SUCCESS", tracer);
-      RetrySettings retrySettings =
-          FAST_RETRY_SETTINGS
-              .toBuilder()
-              .setInitialRetryDelay(Duration.ofMillis(1_000L))
-              .setMaxRetryDelay(Duration.ofMillis(1_000L))
-              .setTotalTimeout(Duration.ofMillis(10_0000L))
-              .build();
+      FailingCallable callable = new FailingCallable(4, "request", "SUCCESS", tracer);
       RetryingExecutorWithContext<String> executor =
           getRetryingExecutor(getAlgorithm(retrySettings, 0, null), localExecutor);
       RetryingFuture<String> future =
           executor.createFuture(callable, FakeCallContext.createDefault().withTracer(tracer));
       callable.setExternalFuture(future);
       future.setAttemptFuture(executor.submit(future));
 
-      Thread.sleep(30L);
+      // The test sleeps a duration long enough to ensure that the future has been submitted for
+      // execution
+      Thread.sleep(50L);
```

Comment:
So the sleep time should be long enough to ensure the future has been submitted, but also short enough so the future is not completed(<275ms in the example)? If that's the case, can we make the retry delay longer and make this sleep time longer? So that we have higher chance of the future being submitted?

Reply:
It is possible to make the retry delay longer, but that doesn't eliminate the possibility that RRD is calculated as multiple consecutive low values (only makes it less likely).

I believe removing Jitter (only for testing) would make the RRD values consistent and guarantee no flakes.

Reply:
Yes the removal of jitter should make it much more stable, but I _think_ there is still a chance that the future is not submitted after 50 ms if the CPU is overwhelmed. 
If I'm understanding correctly, the first retry is at 25ms and the last retry is at 275ms, so the sleep time should between 25 and 275. However. there is no guarantee that the first retry is at 25ms since it's a different thread, so there is a chance that after 50ms, future is not started yet. Let me know if it makes sense.

Reply:
I see. I can a pick a number closer to 175ms (right before the 4th retry occurs) ... say 150ms? Obviously, still no guarantee that it would occur, but would be extremely unlikely the CPU is stuck that long.

I can add in another check the ensure that the future runs at least once.

Reply:
>  I can a pick a number closer to 175ms (right before the 4th retry occurs) ... say 150ms?

If it's too close to the end, then we may have a problem on the other side, that the sleep may take too long so the future is already completed. That's why I was suggesting maybe we can have the retry delay to 1000ms, so that the sleep time 150ms is long enough for 25 ms, and short enough for 1000ms.   

> I can add in another check the ensure that the future runs at least once.

Actually I get this from your [comment](https://github.com/googleapis/sdk-platform-java/pull/2006/files#diff-253797ac36ea1cf440da370eab649ec38de9cf6db2a086857b187a5581d2a0bdR258-R259), I'm not sure if it's worth it to add one. I wouldn't add it if it would make the test more flaky and does not add much value.

Reply:
Ok yeah agreed that 150ms might cut it too close to the other end. I can change the retry multiplier so that the max retry delay will hit 1000ms and buy us enough time.

The reason for the check to ensure it runs once is to check that the future from `executor.submit()` was run. The first attempt is always scheduled to "run immediately" with 0ms delay (should be be able to be scheduled within say a sleep time of 150ms).